### ISSUE TITLE
[Deque Analysis] Item/Entity Homepage "serious" issues

### ISSUE THREAD

**Issue by tdonohue** [DSpace/dspace-angular#1183]
> Serious: This issue results in serious barriers for people with disabilities, and will partially prevent them from accessing fundamental features or content. People relying on assistive technologies will experience significant frustration as a result. Issues falling under this category are major problems, and remediation should be a priority. 

## Deque Analysis Summary

Item / Entity homepages have a total of 4 serious issues (not including color contrast which are handled in other tickets).  These have been grouped into one ticket because they are similar issues on both pages.

1. (ID 470327) "Purpose of the link is not clear in context." on the "Show 4 more" and "Hide last 4" links which appear in long lists of Entity relationships (especially on Person Entity pages like https://demo.dspace.org/entities/person/5a3f7c7a-d3df-419c-b8a2-f00ede62c60a).
    * Fix by one of the following: (1) Provide unique link text that conveys the purpose of each link. OR (2)  Ensure that the link text PLUS its programmatic context conveys the purpose of the link. (e.g. using "aria-label" to provide more details).  See also https://dequeuniversity.com/class/semantic-structure2/links/link-text
2. (ID 471043) ~~"Content is lost at 320px width" on the Item URI/Handle ("URI" field in display) when the link is long, e.g. https://demo.dspace.org/entities/publication/4e203020-8fd6-406a-addc-f4f08365b662~~
     * ~~Fix by ensuring the link also wraps/resizes for smaller views of 320px~~
     * NO LONGER REPRODUCIBLE in 7.6.1. Unsure of when this was fixed.

Full list of issues is viewable at (requires login): https://axeauditor.dequecloud.com/test-run/0856438a-a19a-11eb-bc31-b7d5be387c86/issues?sortField=ordinal&sortDir=asc&filter%5Bseverity%5D=3&filter%5Btype%5D=issue&filter%5Bpage_number%5D=6&filter%5Bpage_number%5D=7

## More Information / Tools

* Consider using [Deque's free Chrome plugin](https://chrome.google.com/webstore/detail/axe-devtools-web-accessib/lhdoppojpmngadmnindnejefpokejbdd?hl=en-US) to check your work

**Comment by artlowel** [comment]
>1. (ID 470327) "Purpose of the link is not clear in context." on the "Show 4 more" and "Hide last 4" links which appear in long lists of Entity relationships (especially on Person Entity pages like https://demo7.dspace.org/entities/person/5a3f7c7a-d3df-419c-b8a2-f00ede62c60a).

We'll add the entity name to the i18n message. It will result in a lot more messages though.

>2. (IDs 471036, 471060) "State of active component lacks 3 to 1 contrast ratio" on the "Full Item page" button.

We could set a contrasting outline on `btn-outline-primary` buttons across DSpace. However this will be a problem for all buttons. The focus outline is set by a bootstrap function to be a color derived from the type of button it is. It makes sense to group [all non-text contrast issues](https://axeauditor.dequecloud.com/test-run/0856438a-a19a-11eb-bc31-b7d5be387c86/issues?activeTab=dt-issue&page=0&pageSize=100&sortField=ordinal&sortDir=asc&row=9&filter%5Bidentifier%5D=1.4.11.b) (or at least the ones where it occurs against the default white body background) in one ticket rather than fixing it in the context of each page separately. Otherwise we'll get slightly different outlines for focus on every page.

>3. (ID 471043) "Content is lost at 320px width" on the Item URI/Handle ("URI" field in display) when the link is long

We'll add the [text-break](https://getbootstrap.com/docs/4.4/utilities/text/#word-break) class to this field

If we leave out 2 this can be done in **2 hours**. 

Fixing 2. for all outlines against the default white background for all button types and form fields will take an estimated **3 hours**

**Comment by tdonohue** [comment]
@artlowel : Thanks for that feedback.  I like your suggestion to break out the non-text contrast issues into a separate ticket (as they are currently embedded in several tickets. So, I'll do that & update the affected tickets.

So, let's leave 2 out of this ticket (I'll remove it once the new ticket exists), and I'll add the estimate of 2hrs & assign this back to you for your team to work on.

**Comment by GraziaQuercia** [comment]
Hi @tdonohue! 4Science will work on this. Our estimation is 4 hours. Thank you!
